### PR TITLE
Introduce a VolumeRenderImpl interface layer

### PR DESCRIPTION
**Pull request by toloudis**
Gather together the common functions for different Volume rendering implementations to be used inside of VolumeDrawable, preparing to add a new one for Agave remote rendering.

### CODE REVIEW

File: src/RayMarchedAtlasVolume.ts [hunk 1]
```diff
@@ -366,4 +367,8 @@ export default class RayMarchedAtlasVolume {
     this.setUniform("textureAtlas", this.channelData.getFusedTexture());
     this.setUniform("textureAtlasMask", this.channelData.maskTexture);
   }
+
+  public setRenderUpdateListener(_listener?: ((iteration: number) => void) | undefined) {
+    0;
```
Comment (ShrimpCryptid) [inline src/RayMarchedAtlasVolume.ts:372]: Any reason for this statement vs. a return?

**Review by ShrimpCryptid — APPROVED** [review]
Looks good to me!

File: src/RayMarchedAtlasVolume.ts (hunk 1 above)
Comment (toloudis) [reply src/RayMarchedAtlasVolume.ts:372]: No special reason. As I recall, typescript didn't want it to be just empty, and putting a 0 seemed like the least amount of typing to have the function do nothing. :).  I know things like lodash provide a `noop` but I figured this did the job without bringing a dependency.  Now that you mention this, `return;` seems a bit more obvious.

File: public/index.ts [hunk 2]
```diff
@@ -1339,11 +1339,12 @@ function main() {
     view3D.setMaxProjectMode(myState.volume, mp);
   };
   renderModeSelect?.addEventListener("change", ({ currentTarget }) => {
-    if ((currentTarget as HTMLOptionElement)!.value === "PT") {
+    const target = (currentTarget as HTMLOptionElement)!;
```
Comment (frasercl) [inline public/index.ts:1342]: Curious how smart typescript and/or babel and/or V8 are about deleting a simple alias like this

File: src/VolumeRenderImpl.ts
Comment (frasercl) [inline src/VolumeRenderImpl.ts:1]: This file could use a run through prettier to fix a bit of inconsistent spacing

**Review by frasercl — APPROVED** [review]
LGTM!

File: src/VolumeRenderImpl.ts
Comment (toloudis) [reply src/VolumeRenderImpl.ts:1]: thanks - not sure why it wasn't autoformatted before.

**Pull request merged**